packages/mongo/tests/changestream_observe_driver_tests.js

```diff
@@ -0,0 +1,123 @@
+      
+      // Wait for removal to propagate
+      await new Promise(resolve => setTimeout(resolve, 100));
+      
+      handle.stop();
+      
+      // Note: We can't easily test the exact counts because the observer
+      // might catch the initial insert or not, depending on timing
+      test.isTrue(changedCount > 0, 'Should have detected changes');
+      test.isTrue(removedCount > 0, 'Should have detected removal');
+      
+    } finally {
+      // Clean up
+      await TestCollection.dropCollectionAsync();
+    }
+  });
+  
+  Tinytest.add('mongo - ChangeStreamObserveDriver - selector filtering', async function (test) {
+    const TestCollection = new Mongo.Collection('test_changestream_filter_' + Random.id());
+    
+    try {
+      let observedDocs = [];
+      
+      // Only observe documents with status: 'active'
+      const handle = TestCollection.find({ status: 'active' }).observe({
+        added: function(doc) {
+          observedDocs.push({ type: 'added', doc: doc });
+        },
+        changed: function(newDoc, oldDoc) {
+          observedDocs.push({ type: 'changed', newDoc: newDoc, oldDoc: oldDoc });
+        },
+        removed: function(oldDoc) {
+          observedDocs.push({ type: 'removed', doc: oldDoc });
+        }
+      });
+      
+      // Wait for observer setup
+      await new Promise(resolve => setTimeout(resolve, 100));
+      
+      // Insert document that matches selector
+      const activeId = await TestCollection.insertAsync({ name: 'active', status: 'active' });
+      
+      // Insert document that doesn't match selector
+      const inactiveId = await TestCollection.insertAsync({ name: 'inactive', status: 'inactive' });
+      
+      // Wait for inserts to propagate
+      await new Promise(resolve => setTimeout(resolve, 200));
+      
+      // Update the active document
+      await TestCollection.updateAsync(activeId, { $set: { name: 'updated_active' } });
+      
+      // Update inactive to active (should trigger add)
+      await TestCollection.updateAsync(inactiveId, { $set: { status: 'active' } });
+      
+      // Wait for updates to propagate
+      await new Promise(resolve => setTimeout(resolve, 200));
+      
+      handle.stop();
+      
+      // We should have observed the active document but not the inactive one initially
+      const addedEvents = observedDocs.filter(event => event.type === 'added');
+      test.isTrue(addedEvents.length >= 1, 'Should have observed at least one added event');
+      
+      // Clean up
+      await TestCollection.removeAsync(activeId);
+      await TestCollection.removeAsync(inactiveId);
+      
+    } finally {
+      await TestCollection.dropCollectionAsync();
+    }
+  });
+  
+} else {
+  // Skip tests if Change Streams are not supported
+  Tinytest.add('mongo - ChangeStreamObserveDriver - skipped (not supported)', function (test) {
+    test.isTrue(true, 'Change Streams not supported in this environment');
+  });
+}
```

Comment:
It’s fine to add this new test for the change stream driver. I’m curious what would happen if we enabled change streams by default across the test suite. If the driver handles everything correctly, all tests should pass. We have many tests expecting publication setups, and the change stream must be included in a way that remains compatible with all current assertions.

To try it, I’d branch off this one and set the env var so all TravisCI test-in-console runs use this change stream driver. Maybe later we can add a CI job that runs change streams for the whole suite.

PD: Current TravisCI tests are failing.